src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -1,16 +1,26 @@
 package seedu.address.logic.parser;
 
+import java.util.stream.Stream;
+
 import seedu.address.logic.commands.Command;
 import seedu.address.logic.parser.exceptions.ParseException;
 
 /**
  * Represents a Parser that is able to parse user input into a {@code Command} of type {@code T}.
  */
-public interface Parser<T extends Command> {
+public abstract class Parser<T extends Command> {
 
     /**
      * Parses {@code userInput} into a command and returns it.
      * @throws ParseException if {@code userInput} does not conform the expected format
      */
-    T parse(String userInput) throws ParseException;
+    public abstract T parse(String userInput) throws ParseException;
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    public boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
nit: would it make more sense for this to be an abstract method that is overridden in each specific parser, so that we don't have to pass the argumentMultiMap in?